Code/Framework/AzToolsFramework/Tests/ComponentModeSwitcherTests.cpp

```diff
@@ -0,0 +1,406 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "ComponentModeTestDoubles.h"
+#include "ComponentModeTestFixture.h"
+
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <AzCore/std/smart_ptr/make_shared.h>
+#include <AzFramework/Viewport/ViewportScreen.h>
+#include <AzTest/AzTest.h>
+#include <AzToolsFramework/ViewportUi/ButtonGroup.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiSwitcher.h>
+#include <AzToolsFramework/ComponentMode/ComponentModeSwitcher.h>
+#include <AzToolsFramework/ViewportUi/ViewportUiManager.h>
+#include <AzToolsFramework/UnitTest/AzToolsFrameworkTestHelpers.h>
+#include <AzToolsFramework/Application/ToolsApplication.h>
+#include <AzToolsFramework/API/EntityCompositionRequestBus.h>
+
+namespace UnitTest
+{
+    using namespace AzToolsFramework;
+    using namespace AzToolsFramework::ComponentModeFramework;
```

Comment:
update: Please remove the use of `using namespace`, it is not recommended to be added to any new code - it can cause all kinds of weird problems due to how it work 🙁 Instead if you want to shorten the names, try using a namespace alias

```
namespace AzTf = AzToolsFramework
```

or use a using declaration

```
using AzToolsFramework::EditorTransformComponentSelectionRequestBus;
```

To bring types into scope (or just use the fully qualified names - this can often be the best thing to do really)